collector/exporter/otelarrowexporter/factory.go

```diff
@@ -48,6 +48,11 @@ func createDefaultConfig() component.Config {
 			Compression: configcompression.Zstd,
 			// We almost read 0 bytes, so no need to tune ReadBufferSize.
 			WriteBufferSize: 512 * 1024,
+			// The `configgrpc` default is pick_first,
+			// which is not great for OTel Arrow exporters
+			// because it concentrates load at a single
+			// destination.
+			BalancerName: "round_robin",
```

Comment:
This is the actual change. When the empty string is passed for `BalancerName`, gRPC ends up deciding `pick_first` in current collector builds, which ends up creating a bunch of streams to the same destination. I'll look into whether there are any popular client-side load balancers we should consider.